### PR TITLE
Clarify challenges

### PR DESCRIPTION
**Pull request by zmb3**
Add some extra info to the challenges to be more explicit as to what we're expecting.

Note: A couple of these commits are formatting changes only. Best to review by commit to not get lost in the formatting noise.

### CODE REVIEW

File: challenges/frontend/challenge.md [hunk 1]
```diff
@@ -131,57 +151,93 @@ This level focuses on the quality of the submission and a comprehensive understa
...
-In case of a negative score result, the hiring manager will contact you and send a list of key observations from the team that affected the result.
+In case of a negative score result, the hiring manager will contact you and send
+a list of key observations from the team that affected the result.
 
 ## Code and project ownership
 
-This is a test challenge and we have no intent of using the code you’ve submitted in production. This is your work, and you are free to do whatever you feel is reasonable with it. In the scenario when you don’t pass, you can open source it with any license and use it as a portfolio project.
+This is a test challenge and we have no intent of using the code you’ve
+submitted in production. This is your work, and you are free to do whatever you
+feel is reasonable with it. In the scenario when you don’t pass, you can open
```
Comment (ibeckermayer) [inline challenges/frontend/challenge.md:202]:
```suggestion
feel is reasonable with it. In the scenario where you don’t pass, you can open
```

File: challenges/frontend/challenge.md [hunk 2]
```diff
@@ -131,57 +151,93 @@ This level focuses on the quality of the submission and a comprehensive understa
...
+source it with any license and use it as a portfolio project.
 
 ## Areas of focus
 
 These are the areas we will be evaluating in the submission:
 
-* Use consistent coding style. We follow recommended [ESLint](https://eslint.org/docs/user-guide/configuring/configuration-files#using-eslintrecommended) and [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react#configuration) rules for JS and use [prettier](https://prettier.io/) code formatter.
+* Use consistent coding style. We follow recommended
```
Comment (ibeckermayer) [inline challenges/frontend/challenge.md:209]:
```suggestion
* Use consistent coding style. We follow the recommended
```

File: challenges/frontend/challenge.md [hunk 3]
```diff
@@ -131,57 +151,93 @@ This level focuses on the quality of the submission and a comprehensive understa
...
+  [ESLint](https://eslint.org/docs/user-guide/configuring/configuration-files#using-eslintrecommended)
+  and
+  [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react#configuration)
+  rules for JS and use [prettier](https://prettier.io/) code formatter.
```
Comment (ibeckermayer) [inline challenges/frontend/challenge.md:213]:
```suggestion
  rules for JS and use the [prettier](https://prettier.io/) code formatter.
```

File: challenges/frontend/challenge.md [hunk 4]
```diff
@@ -131,57 +151,93 @@ This level focuses on the quality of the submission and a comprehensive understa
...
-* Error handling. We pay extra attention to error handling. Make sure that they are properly handled and not ignored.
+* Error handling. We pay extra attention to error handling. Make sure that they
+  are properly handled and not ignored.
 * URL navigation is a big part of great UX, do not ignore it.
-* Keep your CSS simple but not simpler. Do not waste your time on animations, instead make sure that an extra character does not destroy your layout.
-* Make sure that your code is secured and your application is not vulnerable to common web security vulnerabilities. For a senior level, make sure you have a good crypto setup and secure session management.
+* Keep your CSS simple but not simpler. Do not waste your time on animations,
+  instead make sure that an extra character does not destroy your layout.
+* Make sure that your code is secured and your application is not vulnerable to
+  common web security vulnerabilities. For a senior level, make sure you have a
```
Comment (ibeckermayer) [inline challenges/frontend/challenge.md:233]:
```suggestion
  common web security vulnerabilities. The [OWASP Cheat Sheet Series](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html) is a great resource for achieving this. For a senior level, make sure you have a
```

File: challenges/frontend/challenge.md [hunk 5]
```diff
@@ -131,57 +151,93 @@ This level focuses on the quality of the submission and a comprehensive understa
...
+  good crypto setup and secure session management.
 
 ## Scoring
 
-We want to be as transparent as possible on how we will be scoring your submission. The following table describes areas you will be evaluated on. and how they will affect your overall score.
+We want to be as transparent as possible on how we will be scoring your
+submission. The following table describes areas you will be evaluated on. and
```
Comment (ibeckermayer) [inline challenges/frontend/challenge.md:239]:
```suggestion
submission. The following table describes the areas you will be evaluated on, and
```

File: challenges/systems/challenge.md [hunk 6]
```diff
@@ -125,50 +161,85 @@ We are going to compile the program, test it and get back to you.
...
+are going to contribute our time too.
 
-After the final submission, the interview team will assemble and vote using a "+1, -2" anonymous voting system: +1 is submitted whenever a team member accepts the submission, -2 otherwise.
+After the final submission, the interview team will assemble and vote using a
+"+1, -2" anonymous voting system: +1 is submitted whenever a team member accepts
+the submission, -2 otherwise.
 
-In case of a positive result, we will connect you to our HR team who will collect one-two references and will work out other details. You can start the reference collection process in parallel if you would like to speed up the process.
+In case of a positive result, we will connect you to our HR team who will
+collect one-two references and will work out other details. You can start the
```
Comment (ibeckermayer) [inline challenges/systems/challenge.md:195]:
```suggestion
collect one to two references and will work out other details. You can start the
```

File: challenges/frontend/challenge.md [hunk 7]
```diff
@@ -131,57 +151,93 @@ This level focuses on the quality of the submission and a comprehensive understa
...
+stages of your work.
+
+Our team will do their best to provide a high quality review of the submitted
+pull requests in a reasonable time frame. You are spending your time on this, we
+are going to contribute our time too.
+
+After the final submission, the interview team will assemble and vote using +1,
+-2 anonymous voting system: +1 is submitted whenever a team member accepts the
+submission, -2 otherwise. In case of a positive result, we will connect you to
+our HR team who will collect one-two references and will work out other details.
```
Comment (ibeckermayer) [inline challenges/frontend/challenge.md:189]:
```suggestion
our HR team who will collect one to two references and will work out other details.
```

File: challenges/systems/challenge.md [hunk 8]
```diff
@@ -125,50 +161,85 @@ We are going to compile the program, test it and get back to you.
...
-In case of a negative score result, hiring manager will contact you and share a list of the key observations from the team that affected the result.
+In case of a negative score result, hiring manager will contact you and share a
+list of the key observations from the team that affected the result.
 
 ## Code and project ownership
 
-This is a test challenge and we have no intent of using the code you've submitted in production. This is your work, and you are free to do whatever you feel is reasonable with it. In the scenario when you don't pass, you can open source it with any license and use it as a portfolio project.
+This is a test challenge and we have no intent of using the code you've
+submitted in production. This is your work, and you are free to do whatever you
+feel is reasonable with it. In the scenario when you don't pass, you can open
```
Comment (ibeckermayer) [inline challenges/systems/challenge.md:208]:
```suggestion
feel is reasonable with it. In the scenario where you don't pass, you can open
```

File: challenges/systems/challenge.md [hunk 9]
```diff
@@ -125,50 +161,85 @@ We are going to compile the program, test it and get back to you.
...
-* At the minimum, create tests for authentication, networking, and unhappy scenario.
-* Make sure builds are reproducible. Pick any vendoring/packaging system that will allow us to get consistent build results.
-* Ensure error handling and error reporting is consistent. The system should report clear errors and not crash under non-critical conditions.
-* Avoid concurrency and networking errors. Most of the issues we've seen in production are related to data races, networking error handling or goroutine leaks. We will be looking for those errors in your code.
-* Security. Use strong authentication and simplest, but robust authorization. Set up the strongest transport encryption you can. Test it.
+* Use consistent coding style. We follow
+  [Go Coding Style](https://github.com/golang/go/wiki/CodeReviewComments) for
+  the Go language. If you are going to use a different language, please pick
+  coding style guidelines and let us know what they are.
+* At the minimum, create tests for authentication, networking, and unhappy
```
Comment (ibeckermayer) [inline challenges/systems/challenge.md:221]: It's not clear to me what "unhappy scenario" means in this context. Is this meaning to say something like "At a minimum, create at least one test for each of authentication, networking, and an error scenario."?

File: challenges/systems/challenge.md [hunk 10]
```diff
@@ -177,36 +248,62 @@ Use hardcoded values as much as possible and simply add TODO items showing your
...
+Consider making other reasonable trade-offs. Make sure you communicate them to
+the interview team.
 
 Here are some other trade-offs that will help you to spend less time on the task:
 
-Do not implement a system that scales or is highly performing. Describe which performance improvements you would add in the future.
-High availability. It is OK if the system is not highly available. Write down how you would make the system highly available and why your system is not.
-Do not try to achieve full test coverage. This will take too long. Take two key components, e.g. authentication/authorization layer and networking and implement one or two test cases that demonstrate your approach to testing.
+Do not implement a system that scales or is highly performing. Describe which
+performance improvements you would add in the future. High availability. It is
```
Comment (ibeckermayer) [inline challenges/systems/challenge.md:261]:
```suggestion
performance improvements you would add in the future. It is
```

**Review by ibeckermayer — APPROVED** [review]
See my comments for little nitpicks.

File: challenges/systems/challenge.md [hunk 11]
```diff
@@ -125,50 +161,85 @@ We are going to compile the program, test it and get back to you.
...
+  scenario.
+* Make sure builds are reproducible. Pick any vendoring/packaging system that
+  will allow us to get consistent build results.
+* Ensure error handling and error reporting is consistent. The system should
+  report clear errors and not crash under non-critical conditions.
+* Avoid concurrency and networking errors. Most of the issues we've seen in
+  production are related to data races, networking error handling or goroutine
+  leaks. We will be looking for those errors in your code.
+* Security. Use strong authentication and simplest, but robust authorization.
+  Set up the strongest transport encryption you can. Test it.
```
[the remark this answers is not in the document]
Comment (zmb3) [reply challenges/systems/challenge.md:232]: CLI UX was mentioned under the design doc, around line 43. Do you think that's sufficient?

**Review by ravicious — APPROVED** [review]
Good changes here. 👍

**Pull request merged**